onnx/test/new_version_converter_test.py

```diff
@@ -0,0 +1,802 @@
+    def test_Sum(self):    
+        self._test_op_upgrade('Sum', 1, [[2, 3, 4], [2, 3, 4]], [[2, 3, 4]], 
+            attrs={'consumed_inputs': [0]}
+        )
+    def test_Tanh(self):    
+        self._test_op_upgrade('Tanh', 1, attrs={'consumed_inputs': [0]})
+    def test_Tan(self):    
+        self._test_op_upgrade('Tan', 7)
+    def test_TfIdfVectorizer(self):    
+        self._test_op_upgrade('TfIdfVectorizer', 9, [[3]], [[5]], 
+            attrs={'max_gram_length': 3, 'max_skip_count': 1, 'min_gram_length': 2, 'mode': 'TFIDF', 'ngram_counts': [0, 20], 'ngram_indexes': [3, 4]}
+        )
+    def test_ThresholdedRelu(self):    
+        self._test_op_upgrade('ThresholdedRelu', 10)
+    def test_Tile(self):
+        # 5->6 adapter is missing
+        repeats = helper.make_tensor('b', TensorProto.INT64, dims=[3], vals=np.array([1, 2, 3]))
+        self._test_op_upgrade('Tile', 6, [[3, 4, 5], [3]], [[3, 8, 15]], 
+            [TensorProto.FLOAT, TensorProto.INT64], 
+            initializer=[repeats]
+        )
+    def test_TopK(self):    
+        self._test_op_upgrade('TopK', 1, [[3, 4, 5]], [[3, 4, 2], [3, 4, 2]], 
+            output_types=[TensorProto.FLOAT, TensorProto.INT64], 
+            attrs={'k': 2}
+        )
+    def test_Transpose(self):    
+        self._test_op_upgrade('Transpose', 1, [[1, 2, 5, 3, 7]], [[1, 7, 5, 2, 3]], 
+            attrs={'perm': [0, 4, 2, 1, 3]}
+        )
+    def test_Unique_1(self):    
+        self._test_op_upgrade('Unique', 11, [[3, 4, 5]], [[None]])
+    def test_Unique_2(self):    
+        self._test_op_upgrade('Unique', 11, [[3, 4, 5]], [[3, None, 5]], 
+            attrs={'axis': 1}
+        )
+    def test_Unsqueeze(self):    
+        self._test_op_upgrade('Unsqueeze', 1, [[3, 4, 5]], [[3, 4, 1, 5]], attrs={'axes': [2]})
+    # def test_Upsample(self):
+    #     # 6->7 adapter is missing    
+    #     self._test_op_upgrade('Upsample', 7, [[3, 4, 5]], [[6, 6, 10]], 
+    #         attrs={'scales': [2., 1.5, 2.]}
+    #     )
+    def test_Where(self):    
+        self._test_op_upgrade('Where', 9, [[2, 3], [2, 3], [2, 3]], [[2, 3]], 
+            [TensorProto.BOOL, TensorProto.FLOAT, TensorProto.FLOAT]
+        )
+    def test_Xor(self):
+        # 6->7 adapter is missing    
+        self._test_op_upgrade('Xor', 7, [[2, 3], [2, 3]], [[2, 3]], 
+            [TensorProto.BOOL, TensorProto.BOOL], [TensorProto.BOOL]
+        )
+
+    def test_ops_tested(self):
+        all_schemas = onnx.onnx_cpp2py_export.defs.get_all_schemas()
+        all_op_names = [schema.name for schema in all_schemas if schema.domain == '']
+        seq_ops = [
+            'ConcatFromSequence', 
+            'SequenceAt', 
+            'SequenceConstruct', 
+            'SequenceEmpty', 
+            'SequenceErase', 
+            'SequenceInsert', 
+            'SequenceLength', 
+            'SplitToSequence',
+            'NegativeLogLikelihoodLoss',
+            'Scatter',
+            'Upsample'
```

Comment:
I had to (hopefully temporarily) exclude some ops for various reasons:

1) Sequence-based ops can't be properly tested because the version converter replaces all sequence inputs with tensor inputs:
```
import onnx
from onnx import helper, TensorProto, version_converter

seq = helper.make_sequence_value_info('seq', TensorProto.FLOAT, [5, 2])
pos = helper.make_tensor_value_info('pos', TensorProto.INT64, [])
elem = helper.make_tensor_value_info('elem', TensorProto.FLOAT, [5, 2])

node = helper.make_node('SequenceAt', ['seq', 'pos'], ['elem'])

graph = helper.make_graph([node], 'MyGraph', [seq, pos], [elem], [])

model = helper.make_model(
    graph,
    producer_name='test', 
    ir_version=7,
    opset_imports=[helper.make_opsetid('', 11)]
)

converted = version_converter.convert_version(model, 11)
print(model.graph.input[0])
print(converted.graph.input[0])
```
```
name: "seq"
type {
  sequence_type {
    elem_type {
      tensor_type {
        elem_type: 1
        shape {
          dim {
            dim_value: 5
          }
          dim {
            dim_value: 2
          }
        }
      }
    }
  }
}

name: "seq"
type {
  tensor_type {
    elem_type: 0
    shape {
    }
  }
}
```

2) Shape inference crashes on `NegativeLogLikelihoodLoss` (`'Shape inference done'` doesn't get printed):
```
import onnx
from onnx import helper, TensorProto, shape_inference
import numpy as np

in1 = helper.make_tensor_value_info('in1', TensorProto.FLOAT, [3, 4, 5])
in2 = helper.make_tensor_value_info('in2', TensorProto.INT64, [3, 5])
out = helper.make_sequence_value_info('out', TensorProto.FLOAT, [])

node = helper.make_node('NegativeLogLikelihoodLoss', ['in1', 'in2'], ['out'])

graph = helper.make_graph([node], 'MyGraph', [in1, in2], [out], [])

model = helper.make_model(
    graph,
    producer_name='test', 
    ir_version=7,
    opset_imports=[helper.make_opsetid('', 12)]
)

onnx.checker.check_model(model)
print('Check done')
shape_inference.infer_shapes(model)
print('Shape inference done')
```

3) Scatter and Upsample are deprecated (in versions 11 and 10 respectively) and should be turned into different operators, but I couldn't find a way to do it. Is this possible somehow?